src/pages/popup/Popup.tsx

```diff
@@ -76,14 +76,9 @@ export default function Popup(): JSX.Element {
 
   useEffect(() => {
     checkIfVendorDataExists();
+    checkInitialConnection();
   }, []);
 
-  useEffect(() => {
-    if (vendorData) {
-      checkInitialConnection();
```

Comment:
@rodolfomiranda The error Lance discussed about could be due to this. checkInitialConnection was being called twice. Initially and then after setting vendorData. 
There was a race condition between this second call and fetchIdentifiers call.